Forum: Add global `default_view` option (ILIAS 9 Feature)

lscharmer opened the pull request:
This PR applies feature request: https://docu.ilias.de/goto_docu_wiki_wpage_7720_1357.html.

mjansenDatabay commented on Modules/Forum/classes/class.ilObjForumAdministrationGUI.php:
```diff
@@ -92,143 +100,113 @@ public function getAdminTabs(): void
+        $checkbox = fn ($label) => $field->checkbox($this->lng->txt($label), $this->lng->txt($label . '_desc'));
+        $nice = fn ($label) => sprintf('%s (%s)', $this->lng->txt('sort_by_date'), $this->lng->txt($label));
+        $to_string = static fn ($x) => (string) $x;
+        $withOptions = static fn ($x, $options) => array_reduce(
```
New variable names should be `snake_case`.

lscharmer replied:
Done

mjansenDatabay commented:
@lscharmer Could you please resolve the conflict? Thx!

lscharmer committed:
Forum: Add global default_view option.

Apply feature request: https://docu.ilias.de/goto_docu_wiki_wpage_7720_1357.html

lscharmer commented:
@mjansenDatabay the conflicts are resolved.
